bindings/nodejs/src/lib.rs

```diff
@@ -22,94 +22,230 @@ use std::str;
-        Ok(Operator::new(op))
+trait OperatorBuilder {
+    fn make_operator(&self) -> Operator;
+}
+
+impl OperatorBuilder for Memory {
+    fn make_operator(&self) -> Operator {
+        Operator(opendal::Operator::create(opendal::services::Memory::default()).unwrap().finish())
     }
 }
 
-#[allow(dead_code)]
 #[napi]
-pub struct ObjectMeta {
-    pub location: String,
-    pub last_modified: i64,
-    pub size: u32
+pub struct Operator(opendal::Operator);
+
+#[napi]
+impl Operator {
+    #[napi]
+    pub fn object(&self, path: String) -> DataObject {
+        DataObject(self.0.object(&path))
+    }
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
-        Self { inner: op }
+pub struct ObjectMetadata(opendal::ObjectMetadata);
+
+#[napi]
+impl ObjectMetadata {
+    /// Mode of this object.
+    #[napi(getter)]
+    pub fn mode(&self) -> ObjectMode {
+        match self.0.mode() {
+            opendal::ObjectMode::DIR => ObjectMode::DIR,
+            opendal::ObjectMode::FILE => ObjectMode::FILE,
+            opendal::ObjectMode::Unknown => ObjectMode::Unknown,
+        }
+    }
+
+    /// Content-Disposition of this object
+    #[napi(getter)]
+    pub fn content_disposition(&self) -> Option<String> {
+        self.0.content_disposition().map(|s| s.to_string())
+    }
+
+    /// Content Length of this object
+    #[napi(getter)]
+    pub fn content_length(&self) -> Option<u32> {
+        u32::try_from(self.0.content_length()).ok()
+    }
+
+    /// Content MD5 of this object.
+    #[napi(getter)]
+    pub fn content_md5(&self) -> Option<String> {
+        self.0.content_md5().map(|s| s.to_string())
+    }
+
+    /// Content Range of this object.
+    #[napi(getter)]
+    pub fn content_range(&self) -> Option<Vec<u32>> {
+        let content_range = self.0
+            .content_range()
+            .unwrap_or_default();
+        let range = content_range.range().unwrap_or_default();
+        Some(vec![
+            u32::try_from(range.start).ok().unwrap_or_default(),
+            u32::try_from(range.end).ok().unwrap_or_default(),
+            u32::try_from(content_range.size().unwrap_or_default()).ok().unwrap_or_default()
+        ])
+    }
+
+    /// Content Type of this object.
+    #[napi(getter)]
+    pub fn content_type(&self) -> Option<String> {
+        self.0.content_type().map(|s| s.to_string())
+    }
+
+    /// ETag of this object.
+    #[napi(getter)]
+    pub fn etag(&self) -> Option<String> {
+        self.0.etag().map(|s| s.to_string())
+    }
+
+    /// Last Modified of this object.(UTC)
+    #[napi(getter)]
+    pub fn last_modified(&self) -> Option<String> {
+        let (secs, nsecs) = self.0
```

Comment:
We can use [format](https://docs.rs/time/0.3.20/time/struct.OffsetDateTime.html#method.format) and [Rfc3339](https://docs.rs/time/0.3.20/time/format_description/well_known/struct.Rfc3339.html) directly and don't need to depend on `chrono`.